### DISCUSSION TITLE
Footer can't be overridden

### DISCUSSION THREAD

**Discussion by aronquiray** [laravel/nova-issues#3892] [Ideas & Feature Requests] [answered]
- Laravel Version:  8.83.7
- Nova Version: 4.0.5
- PHP Version: 8.0.11
- Database Driver & Version: MySQL
- Operating System and Version: Windows 11
- Browser type and version: Chrome 100.0.4896.75

### Description:
Footer can't be overridden

### Detailed steps to reproduce the issue on a fresh Nova installation:
Nova assets have been published, but seems like it doesn't work. 
optimize:clear , still no effect.

before, it can be overrided via vendor/nova/partials/footer , right now it doesn't. 
![Screenshot_1176](https://user-images.githubusercontent.com/13714092/162249071-b6ef644b-de7a-4a99-adaf-d2ebb3ad769a.jpg)

**Comment by amanpatel** [comment]
Thanks for posting this. I am having a related issue. In Nova 3, we customized the login page to use OAuth server. But since Nova 4 uses vue components for rendering UI, it is not as simple as publishing the views and updating them. I bet the Nova team is probably already aware of this and might have a solution in the next point release.

**Reply by ThaDaVos** [reply]
26 days later and no answer or workaround for this - have you found one? We have the same issue as we are also using a OAuth server (Laravel Passport based)

**Reply by RhysLees** [reply]
> 26 days later and no answer or workaround for this - have you found one? We have the same issue as we are also using a OAuth server (Laravel Passport based)

You can pass your own routes for auth, I use the follow so i can use jetstream auth with 2FA

![image](https://user-images.githubusercontent.com/43909932/166462054-39751e1f-b7d9-4650-8dae-52d87b93e133.png)

**Reply by ThaDaVos** [reply]
Oh haven't noticed you could - probably not in the documentation

**Reply by RhysLees** [reply]
> Oh haven't noticed you could - probably not in the documentation

It's not in the docs, it's an undocumented feature, I found it when I was digging through the source that I could set it

**Comment by aronquiray** [comment]
> Thanks for posting this. I am having a related issue. In Nova 3, we customized the login page to use OAuth server. But since Nova 4 uses vue components for rendering UI, it is not as simple as publishing the views and updating them. I bet the Nova team is probably already aware of this and might have a solution in the next point release.

it's not really a critical bug. however, we don't want our clients to know we're using this haha! anyway, hope it'll resolved be soon.

**Comment by seabasss** [comment]
+1 for wanting to change the footer/header. Thanks!

**Comment by pcorrick** [comment]
I haven't tried it yet but this might help:
https://github.com/Muetze42/nova-assets-changer

**Comment by potsky** [comment]
You can tweak the footer by publishing the (only) `layout.blade.php` view and add something like this in the head section:

```
<!-- Footer -→
<style>
    .mt-8.leading-normal.text-xs.text-gray-500.space-y-1:last-child > p:nth-child(2) {
        display: none;
    }
    .mt-8.leading-normal.text-xs.text-gray-500.space-y-1:last-child > p > a {
        visibility: hidden;
        position: relative;
    }
    .mt-8.leading-normal.text-xs.text-gray-500.space-y-1:last-child > p > a:after {
        visibility: visible;
        position: absolute;
        left:0;
        content: "Whatever";
    }
</style>
```

Not fully useful but it is just an idea...

**Comment by gcg** [comment]
Yeah, this was pretty useful for us as well. We push multiple versions per day and admins knowing the version they use while giving feedback is pretty good so we were keeping our api, laravel and nova versions in the footer for easy access like this: 
![image](https://user-images.githubusercontent.com/811685/162438878-cb9e78bd-c31d-45dc-a68e-7e636df396f3.png)

**Comment by davidhemphill** [comment]
We will probably reintroduce those customization features in the future, once everything settles down a bit.

**Reply by fragkp** [reply]
For a quickfix, it would be cool if you add a custom class/id/attribute to the footer element or something so that we can select these elements easier.

**Reply by michaelhume** [reply]
Or even better to each of the main layout sections - header, sidebar, footer

**Reply by ThaDaVos** [reply]
Any eta on this @davidhemphill ? It's currently stopping a lot of people from upgrading, including us - sadly we only found out about this issue after we bought the unlimited license

**Comment by notdone-au** [comment]
Yes I was disappointed to see this removed from v4 I have a "sensitive" client base and I'm sure the Nova link branding there will cause confusion.

**Reply by aronquiray** [reply]
same, thats why I cannot push this update yet on my clients.

**Comment by clementmas** [comment]
I can't change the favicon anymore on my end. Probably related to the same issue.

Removing these customization features is a deal breaker. I can't upgrade at the moment.

**Reply by aronquiray** [reply]
same! haha.

**Comment by AlexCatch** [comment]
In the meantime, I've had to copy the `laravel/nova` folder from the `vendor` directory and symlink it like v3.0 which allows me to edit the `.vue` files directly, it's not ideal but a simple enough workaround for the time being.

**Comment by wdelfuego** [comment]
I've created a workaround that reintroduces footer content customization to Nova 4.

Installation and usage: [wdelfuego/nova4-customizable-footer](https://github.com/wdelfuego/nova4-customizable-footer)

Let me know if you run into any problems using that package.

**Comment by bbs-smuller** [comment]
Hello,

The same for us ; we bought the unlimited licence and we cannot upgrade all our existing code base because our customer are French and doesn't want to have English wording and un-desired copyrights OR vendors trivia version

Do you have any dates for "reintroduce those customization features" ?

**Comment by leecolarelli** [comment]
Same issue here... our company keeps 99% of the Nova design, but we had changed the sidebar (background, link colors, hover etc) to match our brand, and now won't be using Nova 4 until this is possible.

**Comment by ThaDaVos** [comment]
Like everyone else, running into the same issue regarding customization of Nova 4.x - we have our custom theme for 3.x packaged up and a separate package to allow us to use our OAUTH 2.0 service for logging into our Nova instances - sadly due to the migration to InertiaJs and thus the removal of the Blade files, re-implementing and customizing stuff like this feels like a no-go in Nova 4.x and stops us from upgrading too... wish we knew this before buying the new license

**Reply by RhysLees** [reply]
https://github.com/laravel/nova-issues/discussions/3892#discussioncomment-2680302

**Comment by davidhemphill** [comment, marked as answer]
The footer can now be overridden by using the `Nova::footer` method in your `App\Providers\NovaServiceProvider::boot` method: https://nova.laravel.com/docs/4.0/installation.html#customizing-nova-s-footer

You could return a simple string, HTML, or even the output of `Blade::compile()`. 

We're working on a clean way to allow even more customization of some theme elements, but we'd like to find a way to do it without creating a bunch of issues for folks when they do customize these elements (We got a lot of people who borked their layout and couldn't receive updates in the previous series). In the end, Nova is opinionated, so it may not ever be 1:1 with the previous Blade layout implementation.

**Reply by clementmas** [reply]
Thanks! I believe most people will opt for this one:

```php
Nova::footer(fn () => '');
```

@ Nova maintainers: you guys are very quick to close "issues" and move them to "discussions". I know you want to be productive but it feels dismissive. For instance I don't think this issue should have been closed.

**Reply by michaelhume** [reply]
This is definitely a welcome addition! 

Is anyone else having issues with the provided documentation link? I don't see any relevant information at the target.

I hope there is a road map to further customization - things like modifying the sidebar nav (ex. make is scrollable when there are lots of resource), user nav, etc.. the lack of ability to override feels like a Laravel anti-pattern.

**Reply by wdelfuego** [reply]
@michaelhume yeah, there seems to be some caching issue or something with that link. Click it, then type 'footer' in the search bar in the top right corner and hit Enter to see the relevant documentation.

**Reply by davidhemphill** [reply]
@michaelhume Thanks for bringing the lack of main nav scroll ability to our attention. We'll be pushing a fix for that today!

What do you find you're unable to do with the current user menu customization abilities? What are the "etc" features you're looking for?

**Reply by jcharlesworth** [reply]
If the call to Nova::footer() isn't doing anything, make sure you republish the Nova assets:

`php artisan vendor:publish --tag=nova-assets --force`

I kept seeing the default footer until I did this.

**Reply by michaelhume** [reply]
@davidhemphill - much appreciated!

For the user nav - in v3, you could modify the dropdown. This was useful for replacing the Nova "logout" which would redirect to the original Nova login page. This may be a non-issue now with the _undocumented_ ability to overwrite the login routes. 

"etc" would be more control over the header and logo. Those small changes would go a long way for me personally.

**Reply by davidhemphill** [reply]
> This may be a non-issue now with the undocumented ability to overwrite the login routes.

You're right...changing the login routes might be the solution there.

> "etc" would be more control over the header and logo. 

Can I ask you to be even more specific? Really trying to grasp what people are trying to do that they can't already. 🙂

**Reply by gepopp** [reply]
From my point of view it would make sence to be able to customize the link on the Logo. Using lova to administer a site it would be cool to link the logo on the frontend ...

**Reply by seabasss** [reply]
> From my point of view it would make sence to be able to customize the link on the Logo. Using lova to administer a site it would be cool to link the logo on the frontend ...

I was just going to write exactly this! Link logo to frontend. Right now I manually have to change the url In the browser. This was a simple thing that I liked in Filament.

Also ability to link logo on the Nova login page to the frontend, which Filament didn’t have an option for.

Thanks! Great work guys!

**Reply by RhysLees** [reply]
> > From my point of view it would make sence to be able to customize the link on the Logo. Using lova to administer a site it would be cool to link the logo on the frontend ...
> 
> I was just going to write exactly this! Link logo to frontend. Right now I manually have to change the url In the browser. This was a simple thing that I liked in Filament.
> 
> Also ability to link logo on the Nova login page to the frontend, which Filament didn’t have an option for.
> 
> Thanks! Great work guys!

on nova 3 i just had the site name with the link to front end 
![image](https://user-images.githubusercontent.com/43909932/166987925-f487e07e-619a-4ca3-8b5f-c290ba0f5d90.png)

**Reply by RhysLees** [reply]
> ld be more control over the header and logo. Those small changes woul

I was thinking of making a PR to nova-docs for the route overwriting, is it something worth me doing?

**Reply by michaelhume** [reply]
> From my point of view it would make sence to be able to customize the link on the Logo. Using lova to administer a site it would be cool to link the logo on the frontend ...

This is also something that I had implemented by modifying the user dropdown - adding a "back to app" link.

> Can I ask you to be even more specific? Really trying to grasp what people are trying to do that they can't already. 

Specifically:

- modify the user drop down, ability to add/remove items. 
- modify they logo - ability to 100% control that element, including link, responsive behaviour, alt-text

In reality, I stopped digging into v4 in depth after hitting these initial issues, so this is a pretty cursory list

Thanks for tuning into this issue.

**Reply by davidhemphill** [reply]
You can modify the user menu programmatically: https://nova.laravel.com/docs/4.0/customization/menus.html#appending-prepending-to-the-menu

I use this for a "Visit Website" link.

**Reply by seabasss** [reply]
Thanks! I'd prefer to have the logo link or a link in the header or the side menu rather than hidden in a user menu with an extra click.

I tried to add this to my side menu, but it just opens the app in a modal:

`MenuSection::make('Back to App')->path('../')->icon('external-link'),`

Any way to add an `externalLink` method to `MenuSection` or add ability to have icons on `MenuItem` so they can be on the top level and without indentation? Even just removing the indentation if a `MenuItem` is on the top level would help, icon is a plus.

Thanks!

**Reply by michaelhume** [reply]
> @michaelhume Thanks for bringing the lack of main nav scroll ability to our attention. We'll be pushing a fix for that today!
> 
> What do you find you're unable to do with the current user menu customization abilities? What are the "etc" features you're looking for?

@davidhemphill  - just wanted to ask if this did get implemented, or was still pending? I just tested out the most recent (4.5.1) and in mobile mode the sidebar nav is scrollable, but not with larger displays.

Thanks,

**Reply by davidhemphill** [reply]
I just pushed a fix today will come out in a little bit

**Comment by seabasss** [comment]
Also, I personally don’t need confirm windows on the log out and stop impersonating buttons. They’re just annoying. Remove or option to remove is my vote.

**Reply by RhysLees** [reply]
> Also, I personally don’t need confirm windows on the log out and stop impersonating buttons. They’re just annoying. Remove or option to remove is my vote.

I was gonna make a request to have confirm logout optional i don't mind it on impersonation button

**Reply by seabasss** [reply]
> > Also, I personally don’t need confirm windows on the log out and stop impersonating buttons. They’re just annoying. Remove or option to remove is my vote.
> 
> I was gonna make a request to have confirm logout optional i don't mind it on impersonation button

Is there an official channel for requests, or are threads like this it? Never mind. Found the request tag in discussions.

**Reply by davidhemphill** [reply]
Would be good to add new feature requests in a new discussion so it doesn't get buried here. 🙂
